### ISSUE TITLE
compile Error undefined reference to `mxnet::Operator

### ISSUE THREAD

**Issue by maxenceliu** [dmlc/mxnet#7072]
build/src/operator/instance_norm.o: In function `mxnet::op::InstanceNormProp::CreateOperatorEx(mxnet::Context, std::vector<nnvm::TShape, std::allocator<nnvm::TShape> >*, std::vector<int, std::allocator<int> >*) const':
instance_norm.cc:(.text+0x273): undefined reference to `mxnet::Operator* mxnet::op::CreateOp<mshadow::gpu>(mxnet::op::InstanceNormParam, int)'
build/src/operator/pooling.o: In function `mxnet::op::PoolingProp::CreateOperatorEx(mxnet::Context, std::vector<nnvm::TShape, std::allocator<nnvm::TShape> >*, std::vector<int, std::allocator<int> >*) const':
pooling.cc:(.text+0xfdc): undefined reference to `mxnet::Operator* mxnet::op::CreateOp<mshadow::gpu>(mxnet::op::PoolingParam, int)'
build/src/operator/crop.o: In function `mxnet::op::CropProp::CreateOperator(mxnet::Context) const':
crop.cc:(.text+0x7da): undefined reference to `mxnet::Operator* mxnet::op::CreateOp<mshadow::gpu>(mxnet::op::CropParam)'
build/src/operator/sequence_reverse.o: In function `mxnet::op::SequenceReverseProp::CreateOperatorEx(mxnet::Context, std::vector<nnvm::TShape, std::allocator<nnvm::TShape> >*, std::vector<int, std::allocator<int> >*) const':
sequence_reverse.cc:(.text+0x652): undefined reference to `mxnet::Operator* mxnet::op::CreateOp<mshadow::gpu>(mxnet::op::SequenceReverseParam, int)'
build/src/operator/spatial_transformer.o: In function `mxnet::op::SpatialTransformerProp::CreateOperatorEx(mxnet::Context, std::vector<nnvm::TShape, std::allocator<nnvm::TShape> >*, std::vector<int, std::allocator<int> >*) const':
spatial_transformer.cc:(.text+0x898): undefined reference to `mxnet::Operator* mxnet::op::CreateOp<mshadow::gpu>(mxnet::op::SpatialTransformerParam, int)'
build/src/operator/deconvolution.o: In function `mxnet::op::DeconvolutionProp::CreateOperatorEx(mxnet::Context, std::vector<nnvm::TShape, std::allocator<nnvm::TShape> >*, std::vector<int, std::allocator<int> >*) const':
deconvolution.cc:(.text+0x811): undefined reference to `mxnet::Operator* mxnet::op::CreateOp<mshadow::gpu>(mxnet::op::DeconvolutionParam, int, std::vector<nnvm::TShape, std::allocator<nnvm::TShape> >*, std::vector<nnvm::TShape, std::allocator<nnvm::TShape> >*, mxnet::Context)'
build/src/operator/pad.o: In function `mxnet::op::PadProp::CreateOperatorEx(mxnet::Context, std::vector<nnvm::TShape, std::allocator<nnvm::TShape> >*, std::vector<int, std::allocator<int> >*) const':
pad.cc:(.text+0x14778): undefined reference to `mxnet::Operator* mxnet::op::CreateOp<mshadow::gpu>(mxnet::op::PadParam, int)'
build/src/operator/rnn.o: In function `mxnet::op::RNNProp::CreateOperatorEx(mxnet::Context, std::vector<nnvm::TShape, std::allocator<nnvm::TShape> >*, std::vector<int, std::allocator<int> >*) const':
rnn.cc:(.text+0x70e): undefined reference to `mxnet::Operator* mxnet::op::CreateOp<mshadow::gpu>(mxnet::op::RNNParam, int)'
build/src/operator/concat.o: In function `mxnet::op::ConcatProp::CreateOperatorEx(mxnet::Context, std::vector<nnvm::TShape, std::allocator<nnvm::TShape> >*, std::vector<int, std::allocator<int> >*) const':
concat.cc:(.text+0x3da): undefined reference to `mxnet::Operator* mxnet::op::CreateOp<mshadow::gpu>(mxnet::op::ConcatParam, int)'
build/src/operator/lrn.o: In function `mxnet::op::LocalResponseNormProp::CreateOperatorEx(mxnet::Context, std::vector<nnvm::TShape, std::allocator<nnvm::TShape> >*, std::vector<int, std::allocator<int> >*) const':
lrn.cc:(.text+0x2b7): undefined reference to `mxnet::Operator* mxnet::op::CreateOp<mshadow::gpu>(mxnet::op::LRNParam, int)'
build/src/operator/correlation.o: In function `mxnet::op::CorrelationProp::CreateOperator(mxnet::Context) const':
correlation.cc:(.text+0x3e7): undefined reference to `mxnet::Operator* mxnet::op::CreateOp<mshadow::gpu>(mxnet::op::CorrelationParam)'
build/src/operator/batch_norm_v1.o: In function `mxnet::op::BatchNormV1Prop::CreateOperatorEx(mxnet::Context, std::vector<nnvm::TShape, std::allocator<nnvm::TShape> >*, std::vector<int, std::allocator<int> >*) const':
batch_norm_v1.cc:(.text+0x481): undefined reference to `mxnet::Operator* mxnet::op::CreateOp<mshadow::gpu>(mxnet::op::BatchNormV1Param, int)'
build/src/operator/sequence_mask.o: In function `mxnet::op::SequenceMaskProp::CreateOperatorEx(mxnet::Context, std::vector<nnvm::TShape, std::allocator<nnvm::TShape> >*, std::vector<int, std::allocator<int> >*) const':
sequence_mask.cc:(.text+0x672): undefined reference to `mxnet::Operator* mxnet::op::CreateOp<mshadow::gpu>(mxnet::op::SequenceMaskParam, int)'
build/src/operator/make_loss.o: In function `mxnet::op::MakeLossProp::CreateOperatorEx(mxnet::Context, std::vector<nnvm::TShape, std::allocator<nnvm::TShape> >*, std::vector<int, std::allocator<int> >*) const':
make_loss.cc:(.text+0x606): undefined reference to `mxnet::Operator* mxnet::op::CreateOp<mshadow::gpu>(mxnet::op::MakeLossParam, int)'
build/src/operator/grid_generator.o: In function `mxnet::op::GridGeneratorProp::CreateOperatorEx(mxnet::Context, std::vector<nnvm::TShape, std::allocator<nnvm::TShape> >*, std::vector<int, std::allocator<int> >*) const':
grid_generator.cc:(.text+0x4e5): undefined reference to `mxnet::Operator* mxnet::op::CreateOp<mshadow::gpu>(mxnet::op::GridGeneratorParam, int)'
build/src/operator/pooling_v1.o: In function `mxnet::op::PoolingV1Prop::CreateOperatorEx(mxnet::Context, std::vector<nnvm::TShape, std::allocator<nnvm::TShape> >*, std::vector<int, std::allocator<int> >*) const':
pooling_v1.cc:(.text+0x1115): undefined reference to `mxnet::Operator* mxnet::op::CreateOp<mshadow::gpu>(mxnet::op::PoolingV1Param, int)'
build/src/operator/leaky_relu.o: In function `mxnet::op::LeakyReLUProp::CreateOperator(mxnet::Context) const':
leaky_relu.cc:(.text+0x325): undefined reference to `mxnet::Operator* mxnet::op::CreateOp<mshadow::gpu>(mxnet::op::LeakyReLUParam)'
build/src/operator/identity_attach_KL_sparse_reg.o: In function `mxnet::op::IdentityAttachKLSparseRegProp::CreateOperator(mxnet::Context) const':
identity_attach_KL_sparse_reg.cc:(.text+0x323): undefined reference to `mxnet::Operator* mxnet::op::CreateOp<mshadow::gpu>(mxnet::op::IdentityAttachKLSparseRegParam)'
build/src/operator/upsampling.o: In function `mxnet::op::UpSamplingProp::CreateOperatorEx(mxnet::Context, std::vector<nnvm::TShape, std::allocator<nnvm::TShape> >*, std::vector<int, std::allocator<int> >*) const':
upsampling.cc:(.text+0x1805): undefined reference to `mxnet::Operator* mxnet::op::CreateOp<mshadow::gpu>(mxnet::op::UpSamplingParam, int)'
build/src/operator/svm_output.o: In function `mxnet::op::SVMOutputProp::CreateOperatorEx(mxnet::Context, std::vector<nnvm::TShape, std::allocator<nnvm::TShape> >*, std::vector<int, std::allocator<int> >*) const':
svm_output.cc:(.text+0x696): undefined reference to `mxnet::Operator* mxnet::op::CreateOp<mshadow::gpu>(mxnet::op::SVMOutputParam, int)'
build/src/operator/bilinear_sampler.o: In function `mxnet::op::BilinearSamplerProp::CreateOperatorEx(mxnet::Context, std::vector<nnvm::TShape, std::allocator<nnvm::TShape> >*, std::vector<int, std::allocator<int> >*) const':
bilinear_sampler.cc:(.text+0x6a6): undefined reference to `mxnet::Operator* mxnet::op::CreateOp<mshadow::gpu>(mxnet::op::BilinearSamplerParam, int)'
build/src/operator/softmax_output.o: In function `mxnet::op::SoftmaxOutputProp::CreateOperatorEx(mxnet::Context, std::vector<nnvm::TShape, std::allocator<nnvm::TShape> >*, std::vector<int, std::allocator<int> >*) const':
softmax_output.cc:(.text+0x766a): undefined reference to `mxnet::Operator* mxnet::op::CreateOp<mshadow::gpu>(mxnet::op::SoftmaxOutputParam, int)'
build/src/operator/convolution.o: In function `mxnet::op::ConvolutionProp::CreateOperatorEx(mxnet::Context, std::vector<nnvm::TShape, std::allocator<nnvm::TShape> >*, std::vector<int, std::allocator<int> >*) const':
convolution.cc:(.text+0x7ad): undefined reference to `mxnet::Operator* mxnet::op::CreateOp<mshadow::gpu>(mxnet::op::ConvolutionParam, int, std::vector<nnvm::TShape, std::allocator<nnvm::
▽
TShape> >*, std::vector<nnvm::TShape, std::allocator<nnvm::TShape> >*, mxnet::Context)'
build/src/operator/roi_pooling.o: In function `mxnet::op::ROIPoolingProp::CreateOperatorEx(mxnet::Context, std::vector<nnvm::TShape, std::allocator<nnvm::TShape> >*, std::vector<int, std::allocator<int> >*) const':
roi_pooling.cc:(.text+0xad9): undefined reference to `mxnet::Operator* mxnet::op::CreateOp<mshadow::gpu>(mxnet::op::ROIPoolingParam, int)'
build/src/operator/slice_channel.o: In function `mxnet::op::SliceChannelProp::CreateOperatorEx(mxnet::Context, std::vector<nnvm::TShape, std::allocator<nnvm::TShape> >*, std::vector<int, std::allocator<int> >*) const':
slice_channel.cc:(.text+0x428): undefined reference to `mxnet::Operator* mxnet::op::CreateOp<mshadow::gpu>(mxnet::op::SliceChannelParam, int)'
build/src/operator/fully_connected.o: In function `mxnet::op::FullyConnectedProp::CreateOperatorEx(mxnet::Context, std::vector<nnvm::TShape, std::allocator<nnvm::TShape> >*, std::vector<int, std::allocator<int> >*) const':
fully_connected.cc:(.text+0x4d9): undefined reference to `mxnet::Operator* mxnet::op::CreateOp<mshadow::gpu>(mxnet::op::FullyConnectedParam, int, std::vector<nnvm::TShape, std::allocator<nnvm::TShape> >*, std::vector<nnvm::TShape, std::allocator<nnvm::TShape> >*, mxnet::Context)'
build/src/operator/regression_output.o: In function `mxnet::op::RegressionOutputProp<(mxnet::op::reg_enum::RegressionOutputType)1>::CreateOperator(mxnet::Context) const':
regression_output.cc:(.text._ZNK5mxnet2op20RegressionOutputPropILNS0_8reg_enum20RegressionOutputTypeE1EE14CreateOperatorENS_7ContextE[_ZNK5mxnet2op20RegressionOutputPropILNS0_8reg_enum20RegressionOutputTypeE1EE14CreateOperatorENS_7ContextE]+0x13): undefined reference to `mxnet::Operator* mxnet::op::CreateRegressionOutputOp<mshadow::gpu>(mxnet::op::reg_enum::RegressionOutputType, mxnet::op::RegressionOutputParam)'
build/src/operator/regression_output.o: In function `mxnet::op::RegressionOutputProp<(mxnet::op::reg_enum::RegressionOutputType)2>::CreateOperator(mxnet::Context) const':
regression_output.cc:(.text._ZNK5mxnet2op20RegressionOutputPropILNS0_8reg_enum20RegressionOutputTypeE2EE14CreateOperatorENS_7ContextE[_ZNK5mxnet2op20RegressionOutputPropILNS0_8reg_enum20RegressionOutputTypeE2EE14CreateOperatorENS_7ContextE]+0x13): undefined reference to `mxnet::Operator* mxnet::op::CreateRegressionOutputOp<mshadow::gpu>(mxnet::op::reg_enum::RegressionOutputType, mxnet::op::RegressionOutputParam)'
build/src/operator/regression_output.o: In function `mxnet::op::RegressionOutputProp<(mxnet::op::reg_enum::RegressionOutputType)0>::CreateOperator(mxnet::Context) const':
regression_output.cc:(.text._ZNK5mxnet2op20RegressionOutputPropILNS0_8reg_enum20RegressionOutputTypeE0EE14CreateOperatorENS_7ContextE[_ZNK5mxnet2op20RegressionOutputPropILNS0_8reg_enum20RegressionOutputTypeE0EE14CreateOperatorENS_7ContextE]+0x10): undefined reference to `mxnet::Operator* mxnet::op::CreateRegressionOutputOp<mshadow::gpu>(mxnet::op::reg_enum::RegressionOutputType, mxnet::op::RegressionOutputParam)'
build/src/operator/l2_normalization.o: In function `mxnet::op::L2NormalizationProp::CreateOperator(mxnet::Context) const':
l2_normalization.cc:(.text+0x26d): undefined reference to `mxnet::Operator* mxnet::op::CreateOp<mshadow::gpu>(mxnet::op::L2NormalizationParam)'
build/src/operator/activation.o: In function `mxnet::op::ActivationProp::CreateOperatorEx(mxnet::Context, std::vector<nnvm::TShape, std::allocator<nnvm::TShape> >*, std::vector<int, std::allocator<int> >*) const':
activation.cc:(.text+0xdc4): undefined reference to `mxnet::Operator* mxnet::op::CreateOp<mshadow::gpu>(mxnet::op::ActivationParam, int, nnvm::TShape const&)'
build/src/operator/batch_norm.o: In function `mxnet::op::BatchNormProp::CreateOperatorEx(mxnet::Context, std::vector<nnvm::TShape, std::allocator<nnvm::TShape> >*, std::vector<int, std::allocator<int> >*) const':
batch_norm.cc:(.text+0x7cd): undefined reference to `mxnet::Operator* mxnet::op::CreateOp<mshadow::gpu>(mxnet::op::BatchNormParam, int, nnvm::TShape const&)'
build/src/operator/dropout.o: In function `mxnet::op::DropoutProp::CreateOperatorEx(mxnet::Context, std::vector<nnvm::TShape, std::allocator<nnvm::TShape> >*, std::vector<int, std::allocator<int> >*) const':
dropout.cc:(.text+0x48b): undefined reference to `mxnet::Operator* mxnet::op::CreateOp<mshadow::gpu>(mxnet::op::DropoutParam, int)'
build/src/operator/softmax_activation.o: In function `mxnet::op::SoftmaxActivationProp::CreateOperator(mxnet::Context) const':
softmax_activation.cc:(.text+0xfbc): undefined reference to `mxnet::Operator* mxnet::op::CreateOp<mshadow::gpu>(mxnet::op::SoftmaxActivationParam)'
build/src/operator/convolution_v1.o: In function `mxnet::op::ConvolutionV1Prop::CreateOperatorEx(mxnet::Context, std::vector<nnvm::TShape, std::allocator<nnvm::TShape> >*, std::vector<int, std::allocator<int> >*) const':
convolution_v1.cc:(.text+0xa6d): undefined reference to `mxnet::Operator* mxnet::op::CreateOp<mshadow::gpu>(mxnet::op::ConvolutionV1Param, int, std::vector<nnvm::TShape, std::allocator<nnvm::TShape> >*, std::vector<nnvm::TShape, std::allocator<nnvm::TShape> >*, mxnet::Context)'
build/src/operator/swapaxis.o: In function `mxnet::op::SwapAxisProp::CreateOperatorEx(mxnet::Context, std::vector<nnvm::TShape, std::allocator<nnvm::TShape> >*, std::vector<int, std::allocator<int> >*) const':
swapaxis.cc:(.text+0x7ba): undefined reference to `mxnet::Operator* mxnet::op::CreateOp<mshadow::gpu>(mxnet::op::SwapAxisParam, int)'
build/src/operator/sequence_last.o: In function `mxnet::op::SequenceLastProp::CreateOperatorEx(mxnet::Context, std::vector<nnvm::TShape, std::allocator<nnvm::TShape> >*, std::vector<int, std::allocator<int> >*) const':
sequence_last.cc:(.text+0x692): undefined reference to `mxnet::Operator* mxnet::op::CreateOp<mshadow::gpu>(mxnet::op::SequenceLastParam, int)'
collect2: error: ld returned 1 exit status
make: *** [bin/im2rec] Error 1

**Comment by madjam** [comment]
Can you please fill out new issue template (pasted below). The will help you get a answer faster.

---

For bugs or installation issues, please provide the following information.
The more information you provide, the more likely people will be able to help you.

## Environment info
Operating System:

Compiler:

Package used (Python/R/Scala/Julia):

MXNet version:

Or if installed from source:

MXNet commit hash (`git rev-parse HEAD`):

If you are using python package, please provide

Python version and distribution:

If you are using R package, please provide

R `sessionInfo()`:

## Error Message:
Please paste the full error message, including stack trace.

## Minimum reproducible example
if you are using your own code, please provide a short script that reproduces the error.

## Steps to reproduce
or if you are running standard examples, please provide the commands you have run that lead to the error.

1.
2.
3.

## What have you tried to solve it?

1.
2.
3.

**Comment by maxenceliu** [comment]
I think I have solved it , by removing the superfluous blank in the line USE_CUDA = 1 (in the file of config.mk)
